### PR TITLE
Start filter stuff

### PR DESCRIPTION
**Pull request by stdavis**
This is a first draft on how we could do the filter stuff. It only filters the station points on the map at this point. Nothing in the grid yet.

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -28,11 +29,15 @@
...
     "tagname",
     "tailwindcss",
+    "tanstack",
     "topo",
+    "UDWR",
     "ugrc",
     "usgs",
     "vite",
-    "wkid"
+    "WILDADMIN",
```
Comment (steveoh) [inline .vscode/settings.json:38]: We probably don't want this in here

File: src/components/Filter.tsx [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+  const stationsLayer = useRef<FeatureLayer>();
+  const { filter } = useFilter();
+
+  useEffect(() => {
+    if (!mapView || !addLayers) {
+      return;
+    }
+
+    stationsLayer.current = new FeatureLayer({
+      url: 'https://wrimaps.utah.gov/arcgis/rest/services/Electrofishing/Public/MapServer/0',
```
Comment (steveoh) [inline src/components/Filter.tsx:22]: Is this ok to be static or will it need to come from an environment?

File: src/main.tsx [hunk 3]
```diff
@@ -19,14 +21,34 @@ if (import.meta.env.VITE_FIREBASE_CONFIG) {
   firebaseConfig = JSON.parse(import.meta.env.VITE_FIREBASE_CONFIG);
 }
 
+const MainErrorFallback = ({ error, resetErrorBoundary }: { error: Error; resetErrorBoundary: () => void }) => {
+  return (
+    <div className="static flex h-screen w-screen items-center justify-center">
+      <div className="flex-col items-center">
+        <h1>Something went wrong</h1>
+        <pre style={{ color: 'red' }}>{error.message}</pre>
```
Comment (steveoh) [inline src/main.tsx:29]: Replace style with classname

File: src/ErrorFallback.tsx [hunk 4]
```diff
@@ -0,0 +1,8 @@
+export default function ErrorFallback({ error }: { error: Error }) {
+  return (
+    <div role="alert">
+      <p>Something went wrong:</p>
+      <pre style={{ color: 'red' }}>{error.message}</pre>
```
Comment (steveoh) [inline src/ErrorFallback.tsx:5]: Let's use tailwind here for consistency

**Review by steveoh — COMMENTED** [review]
Wow, great progress. I skimmed it on my phone and I'm pretty happy where you're going.

File: src/components/Filter.tsx (hunk 2 above)
Comment (stdavis) [reply src/components/Filter.tsx:22]: This is a temporary use of the public service to allow me to work on the filters until we get UtahID wired up. I'll add a comment.

**Comment by stdavis** [comment]
Do you want to take another look and work towards a merge or should I continue working on this branch?

**Comment by steveoh** [comment]
> Do you want to take another look and work towards a merge or should I continue working on this branch?

Yep, will do. Will you update the readme for development so we're both on the same page.

File: README.md [hunk 5]
```diff
@@ -25,7 +25,10 @@ Production: [https://electrofishing-query.ugrc.utah.gov](https://electrofishing-
 
 ### Setup
 
-1. `npm install` from root and functions folders.
-1. `npm run copy:arcgis` to copy the ArcGIS API for JavaScript files to the public folder. This needs to be done anytime the arcgis package is updated.
-1. Create `.env.local` files in root and functions folders using the corresponding `.env` files as examples.
-1. `npm start` to start the development server.
```
Comment (stdavis) [inline README.md]: It would have been easier to just write this yourself rather than asking me to do it and then changing it all. :)

Comment (steveoh) [reply README.md]: What you added was great. I unnecessarily formatted it. Sorry about that.

**Pull request merged**